pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -186,8 +186,12 @@ func IsKRM(n *yaml.RNode) error {
 
 func AreKRM(nodes []*yaml.RNode) error {
 	for i := range nodes {
+		// don't do this check for .json files
+		path, _, _ := kioutil.GetFileAnnotations(nodes[i])
+		if strings.HasSuffix(path, ".json") {
```

Comment:
why are we excluding `.json` files ? json can contain KRM resources.. right ?
Is it motivated by the use-case that involves schema files as `.json` ?

Reply:
`TestSinkCommandJSON` failed without this check. I'm not sure if the readers and writers in kyaml are intended to be used for JSON.

Reply:
`TestSourceCommandJSON` and `TestSinkCommandJSON` failed without this check. I'm not sure if the readers and writers in kyaml are intended to be used for JSON.

Reply:
resources can exist in `json` and `yaml`, so we need to be enforcing on both. But I remember cases where people are using `schema` files in packages and they will be non `non-KRM`. 

I have seen `source` and `sink` being used to stage or prep-up the kpt packages, so these might deal with non-KRM resources, so I am thinking if we should be enforcing KRM checks in `source` and `sink`. sort of keeping them dumb.

Reply:
Can you show the use case using concrete commands?

Note on JSON vs YAML: Do we actually have e2e tests that use JSON. I'm not sure if anyone uses JSON manifests, but we shouldn't claim it works unless it's tested.

Reply:
> But I remember cases where people are using schema files in packages and they will be non non-KRM.

If we enforce on json, the use case described in https://github.com/GoogleContainerTools/kpt/issues/2144 will break (they are using a JSON OpenAPI schema file as input to kustomize)